examples/player/simid_player.js

```diff
@@ -240,67 +255,111 @@ class SimidPlayer {
   }
 
   /**
-   * Returns the original specified dimensions of the non-linear creative.
+   * Checks whether the input dimensions are valid and fit in the player window.
+   * @param {!Object} dimensions A dimension that contains x, y, width & height fields.
+   * @return {boolean}
+   */
+  isValidDimensions_(dimensions) {
+    const playerDiv = document.getElementById('player_div');
+    const playerRect = playerDiv.getBoundingClientRect();
+
+    const heightFits = parseInt(dimensions.y) + parseInt(dimensions.height) <= parseInt(playerRect.height);
+    const widthFits = parseInt(dimensions.x) + parseInt(dimensions.width) <= parseInt(playerRect.width);
+    
+    return heightFits && widthFits;
+  }
+
+  /**
+   * Returns the specified dimensions of the non-linear creative.
    * @return {!Object}
    */
-  getNonLinearDimensions() {
-    const x_val = document.getElementById('x_val').value;
-    const y_val = document.getElementById('y_val').value;
-    const width = document.getElementById('width').value;
-    const height = document.getElementById('height').value;
+  getCreativeDimensions_() {
+    let newDimensions = {};
+    newDimensions.x = document.getElementById('x_val').value;
+    newDimensions.y = document.getElementById('y_val').value;
+    newDimensions.width = document.getElementById('width').value;
+    newDimensions.height = document.getElementById('height').value;
 
-    return {
-      'x' : x_val,
-      'y' : y_val,
-      'width' : width,
-      'height' : height,
-    };
+    return newDimensions;
+  }
+
+  /**
+   * Changes the simid iframe dimensions to the given dimensions
+   * @param {!Object} A dimension that contains an x,y,width & height fields.
+   */
+  setSimidIframeDimensions_(resizeDimensions) {
+    this.simidIframe_.style.height = resizeDimensions.height;
+    this.simidIframe_.style.width = resizeDimensions.width;
+    this.simidIframe_.style.left = `${resizeDimensions.x}px`;
+    this.simidIframe_.style.top = `${resizeDimensions.y}px`;
+  }
+
+  /** The creative wants to expand the ad. */
+  onExpandResize(incomingMessage) {
+    if (this.isLinearAd_) {
+      this.simidProtocol.reject(incomingMessage, "Cannot expand linear ads");
```

Comment:
Instead of a string use an object.
https://interactiveadvertisingbureau.github.io/SIMID/#protocol-message-categories-reject